### PR TITLE
Feat(rpc): Implement eth_call by transaction simulation

### PR DESCRIPTION
**Pull request by bulenttastan**
### Description
`eth_call` implementation by simulating an entry function or script transaction. Part of #47.

### Changes
- `call_transaction` method to execute entry fn or script input
- `deploy_contract` helper test method to set up `eth_call` test
- Tests for entry fn and script bytecode inputs

### Testing
✓ New tests for entry fn and script

### Notes
Refactored `simulate_transaction` into `simulate.rs` under `move_execution`.

### CODE REVIEW

File: moved/src/move_execution/simulate.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        &B256::random(),
+        state,
+        genesis_config,
+        0,
+        base_token,
+        block_header,
+    )
+}
+
+pub fn call_transaction(
```
Comment (birchmd) [inline moved/src/move_execution/simulate.rs:61]: This function currently skips the nonce check which is present in `execute_transaction`. We could add it in, but that would be duplicating some code. And in the future when there are more pre-/post-transaction operations (e.g. charging and refunding gas) then those operations would need to be duplicated here as well.

I understand the point of having this function separate is to enable getting return values for view calls, but I think some refactoring will be necessary to have this function reuse logic from `execute_transaction`. Probably the easiest way to do that is to factor out the pre-transaction check as another function which we can use here as well as in `execute_transaction`. Similarly, we need to make sure this function gets the gas charging functionality when it gets implemented in the future.

**Review by birchmd — COMMENTED** [review]
This looks pretty good overall! I just have a few comments/questions.

File: moved/src/move_execution/simulate.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    request: TransactionRequest,
+    state: &(impl MoveResolver<PartialVMError> + TableResolver),
+) -> crate::Result<Vec<u8>> {
+    let tx = NormalizedEthTransaction::from(request.clone());
```
[the remark this answers is not in the document]
Comment (bulenttastan) [reply moved/src/move_execution/simulate.rs:65]: It fails without `.clone()`, complains that request doesn't derive `Copy`. I change it to construct `tx` after getting the nonce.

File: moved/src/move_execution/simulate.rs (hunk 1 above)
Comment (bulenttastan) [reply moved/src/move_execution/simulate.rs:62]: Working on this.
I'll separate the transaction metadata checks from `canonical` and use it in both places.

**Review by birchmd — APPROVED** [review]

**Pull request merged**